NDTensors/src/default_storage.jl

```diff
@@ -0,0 +1,39 @@
+## This is a fil which specifies the default storage type provided some set of parameters
+## The parameters are the element type and storage type
+function default_storage_type(
+  storetype::Type{<:TensorStorage}=Dense,
+  eltype::Type=Float64,
+  datatype::Type{<:AbstractArray}=Vector,
+  index=Tuple{Int},
+)
+  #TODO check if indices are block sparse or not
+  type = nothing
+  try
+    type = storetype{eltype,datatype{eltype}}
+  catch e
+    println("Cannot construct storetype", storetype)
+    throw(e)
+  end
+  return type
+end
+
+default_storage_type(eltype::Type{<:Number}) = default_storage_type(Dense, eltype)
+
+function default_storage_type(datatype::Type{<:AbstractArray})
+  return default_storage_type(Dense, Float64, datatype)
```

Comment:
Here I would use the existing `set_datatype` function, for example:
```julia
default_storagetype(datatype::Type{<:AbstractArray}) = set_datatype(Dense, datatype)
```

Reply:
Then it is expected that `set_datatype` is defined for each storage type, which is already expected by my implementation of `adapt`.

Possibly we can think about making a default fallback version of `set_datatype` for all storage types using Julia introspection tools, see for example: https://discourse.julialang.org/t/error-calling-similar-array-complexf32-float32/86248 and https://discourse.julialang.org/t/manipulating-parametric-types/6893, however it doesn't seem like there is an official Julia API for manipulating type parameters of a type.